demos/shadow/shadow_demo_main/shadow_demo_main.c

```diff
@@ -0,0 +1,644 @@
+            }
+        }
+        else
+        {
+            LogError( ( "No version in json document!!\n\n" ) );
+        }
+    }
+    else
+    {
+        LogError( ( "The json document is invalid!!\n\n" ) );
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+static void updateAcceptedHandler( MQTTPublishInfo_t * pPublishInfo )
+{
+    char * outValue = NULL;
+    uint32_t outValueLength = 0U;
+    uint32_t receivedToken = 0U;
+
+    assert( pPublishInfo != NULL );
+    assert( pPublishInfo->pPayload != NULL );
+
+    LogInfo( ( "/update/accepted json payload:%s.\n\n", pPublishInfo->pPayload ) );
+
+    /* Handle the reported state with state change in /update/accepted topic. */
+    /* Thus we will retrieve the client token from the json document to see if
+     * it's the same one we sent with reported state on the /update topic.
+     */
+    if ( JSONSuccess == JSON_Validate( pPublishInfo->pPayload,
+                                       pPublishInfo->payloadLength ) )
+    {
+        /*
+         * The payload will look similar to this:
+         *  {
+         *      "state": {
+         *          "reported": {
+         *          "powerOn": 1
+         *          }
+         *      },
+         *      "metadata": {
+         *          "reported": {
+         *          "powerOn": {
+         *              "timestamp": 1596573647
+         *          }
+         *          }
+         *      },
+         *      "version": 14698,
+         *      "timestamp": 1596573647,
+         *      "clientToken": "022485"
+         *  }
+         */
+
+        JSONStatus_t result = JSONSuccess;
+
+        /* Get clientToken from json documents. */
+        result = JSON_Search( ( char * ) pPublishInfo->pPayload,
+                              pPublishInfo->payloadLength,
+                              "clientToken",
+                              sizeof("clientToken") - 1,
+                              '.',
+                              & outValue,
+                              ( size_t * ) & outValueLength );
+
+        if ( result == JSONSuccess )
+        {
+            LogInfo( ( "clientToken: %.*s\n\n", outValueLength,
+                                                outValue ) );
+
+            /* Convert the code to an unsigned integer value. */
+            receivedToken = ( uint32_t ) strtoul( outValue, NULL, 10 );
+
+            LogInfo( ( "receivedToken:%d, clientToken:%u \r\n", receivedToken,  clientToken) );
+
+            /* If the clientToken in this update/accepted message matches the one we
+             * published before, it means the device shadow has accepted our latest
+             * reported state. We are done. */
+            if ( receivedToken == clientToken )
+            {
+                LogInfo( ( "Received response from the device shadow. Previously published "
+                           "update with clientToken=%u has been accepted. \n\n", clientToken) );
+            }
```

Comment:
else log a debug message.

Reply:
will add it.

Reply:
This [commit](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1104/commits/49b40ee1665d75b8ed186c326788b6200655e313) should fix the comment.
please let me know if you have other questions.